examples/microprofile/multiport/src/test/java/io/helidon/examples/microprofile/multiport/MainTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.examples.microprofile.multiport;
+
+import javax.inject.Inject;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import io.helidon.microprofile.tests.junit5.HelidonTest;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Unit test for {@link Main}.
+ */
+@HelidonTest
+class MainTest {
```

Comment:
Testing should never use set ports, this should use random ports. 
Right now, you can only use `HelidonTest` to inject a client pointing to the main port (which is randomized by the test itself).
To be able to use additional ports, we would need to either update the `HelidonTest` implementation to support additional clients, or add injection support for `WebServer` in Helidon MP.

Reply:
Fixed.
For now I explicitly set the config to dynamically generate ports and inject `ServerCdiExtension` to access the port numbers of the running server (this required adding` ServerCdiExtension.getPort(String)` which was missing).